go-controller/pkg/ovn/master.go

```diff
@@ -831,6 +850,21 @@ func (oc *DefaultNetworkController) deleteNodeEvent(node *kapi.Node) error {
 	if err := oc.deleteNode(node.Name); err != nil {
 		return err
 	}
+
+	// Some nodes might not have the chassisID annotation yet so trying to
+	// parse it can fail.  That's not a problem.  Either they don't need
+	// a chassisID one or it wasn't there yet before the node got deleted.
+	// In any case, we have not created a chassis template record for that
+	// node.
+	//
+	// We cannot remove Chassis_Template_Var records within deleteNode()
+	// because we need to look them up by chassisID and not by node hostname.
+	if chassisID, err := util.ParseNodeChassisIDAnnotation(node); err == nil {
+		if err = libovsdbops.DeleteChassisTemplateVar(oc.nbClient, &nbdb.ChassisTemplateVar{Chassis: chassisID}); err != nil {
+			return fmt.Errorf("failed deleting chassis template variables for %s: %v", node.Name, err)
+		}
```

Comment:
I think it would be good if we actually moved this to deleteNode. I see that it is called form some other place and also it might be called from some other places in the future, and it would make sense the we would delete ChassisTemplateVar within that method just as we delete Chassis.

One alternative would be to pass the kapi.Node into deleteNode, not just the name.

Another trick you can do in `deleteNode()` where you have:
```
	p := func(item *sbdb.Chassis) bool {
		return item.Hostname == nodeName
	}
	if err := libovsdbops.DeleteChassisWithPredicate(oc.sbClient, p); err != nil {
		return fmt.Errorf("failed to remove the chassis associated with node %s in the OVN SB Chassis table: %v", nodeName, err)
	}
```
You can pick up the chassis name from that predicate and delete template var with it:
```
	var chassisName string
        p := func(item *sbdb.Chassis) bool {
        if item.Hostname == nodeName {
            chassisName = item.Name
            return true
        }
		return false
	}
	if err := libovsdbops.DeleteChassisWithPredicate(oc.sbClient, p); err != nil {
		return fmt.Errorf("failed to remove the chassis associated with node %s in the OVN SB Chassis table: %v", nodeName, err)
	}
        if err = libovsdbops.DeleteChassisTemplateVar(oc.nbClient, &nbdb.ChassisTemplateVar{Chassis: chassisName}); err != nil {
		return fmt.Errorf("failed deleting chassis template variables for %s: %v", node.Name, err)
	}
```
watch out though for multiple matches or no matches (in case chassisName needs to be an array or we need to check ends up being empty)

Reply:
Good idea, I added now something along these lines.